linux/shim/io.go

```diff
@@ -15,8 +15,9 @@ import (
 )
 
 func copyConsole(ctx context.Context, console console.Console, stdin, stdout, stderr string, wg, cwg *sync.WaitGroup) error {
+	fctx := context.Background()
 	if stdin != "" {
-		in, err := fifo.OpenFifo(ctx, stdin, syscall.O_RDONLY, 0)
+		in, err := fifo.OpenFifo(fctx, stdin, syscall.O_RDONLY, 0)
```

Comment:
Not sure I get the meaning of that change. If context is closed while the `open(2)` is blocking it already closes it down. `OpenFifo` should definitely be used with contexts that do cancel, otherwise they will eventually get stuck. Not sure I understand how https://github.com/containerd/fifo/blob/master/fifo.go#L77 could be reached because if `O_NONBLOCK` is not set `OpenFifo` will not return before the file is actually opened and this select will never return from that line.